The unit test `test_build_multi_rate_fir_filter` is failing

The unit test `test_build_multi_rate_fir_filter` is currently failing, likely due to a change in SciPy’s `remez()` function, which is invoked by ISCE3 [here](https://github.com/isce-framework/isce3/blob/968e78fb90e09fd69d34770fc6ab238ac2b0fdcb/python/packages/isce3/signal/fir_filter_func.py#L79). Below is the CI error message:

```
../../../../../../tests/python/packages/isce3/signal/fir_filter_func.py:121: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../../../install/packages/isce3/signal/fir_filter_func.py:395: in build_multi_rate_fir_filter
    coefs = cheby_equi_ripple_filter(
../../../../../../install/packages/isce3/signal/fir_filter_func.py:79: in cheby_equi_ripple_filter
    coeffs = spsg.remez(len_flt, 0.5 / samprate
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

numtaps = 111, bands = array([0.        , 0.08333333, 0.1       , 0.5       ])
desired = array([1., 0.])

    def remez(numtaps, bands, desired, *, weight=None, type='bandpass',
              maxiter=25, grid_density=16, fs=None):
        """
        Calculate the minimax optimal filter using the Remez exchange algorithm.
    
        Calculate the filter-coefficients for the finite impulse response
        (FIR) filter whose transfer function minimizes the maximum error
        between the desired gain and the realized gain in the specified
        frequency bands using the Remez exchange algorithm.
    
        Parameters
        ----------
        numtaps : int
            The desired number of taps in the filter. The number of taps is
            the number of terms in the filter, or the filter order plus one.
        bands : array_like
            A monotonic sequence containing the band edges.
            All elements must be non-negative and less than half the sampling
            frequency as given by `fs`.
        desired : array_like
            A sequence half the size of bands containing the desired gain
            in each of the specified bands.
        weight : array_like, optional
        This example shows a band-pass filter with a pass-band from 2 kHz to
        5 kHz.  The transition width is 260 Hz and the length of the filter
        is 63, which is smaller than in the other examples:
    
        >>> band = [2000, 5000]  # Desired pass band, Hz
        >>> trans_width = 260    # Width of transition from pass to stop, Hz
        >>> numtaps = 63         # Size of the FIR filter.
        >>> edges = [0, band[0] - trans_width, band[0], band[1],
        ...          band[1] + trans_width, 0.5*fs]
        >>> taps = signal.remez(numtaps, edges, [0, 1, 0], fs=fs)
        >>> w, h = signal.freqz(taps, [1], worN=2000, fs=fs)
        >>> plot_response(w, h, "Band-pass Filter")
        >>> plt.show()
    
        The low order leads to higher ripple and less steep transitions.
    
        The next example shows a band-stop filter.
    
        >>> band = [6000, 8000]  # Desired stop band, Hz
        >>> trans_width = 200    # Width of transition from pass to stop, Hz
        >>> numtaps = 175        # Size of the FIR filter.
        >>> edges = [0, band[0] - trans_width, band[0], band[1],
        ...          band[1] + trans_width, 0.5*fs]
        >>> taps = signal.remez(numtaps, edges, [1, 0, 1], fs=fs)
        >>> w, h = signal.freqz(taps, [1], worN=2000, fs=fs)
        >>> plot_response(w, h, "Band-stop Filter")
        >>> plt.show()
    
        """
        xp = array_namespace(bands, desired, weight)
        bands = np.asarray(bands)
        desired = np.asarray(desired)
>       if weight:
           ^^^^^^
E       ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

/home/runner/micromamba/envs/isce3/lib/python3.12/site-packages/scipy/signal/_fir_filter_design.py:837: ValueError
____________ test_bandpass[0.4166666666666667--0.192[708](https://github.com/isce-framework/isce3/actions/runs/15982576518/job/45080120442?pr=39#step:11:709)33333333334] ____________
```

@rad-eng-59 , is this error familiar to you?

--- a/python/packages/isce3/signal/fir_filter_func.py
+++ b/python/packages/isce3/signal/fir_filter_func.py
@@ -78,8 +78,8 @@ def cheby_equi_ripple_filter(samprate, bandwidth, rolloff=1.2, ripple=0.1,
     # get LPF coeffs
     coeffs = spsg.remez(len_flt, 0.5 / samprate
                         * np.array([0, bandwidth, fstop, samprate]),
-                        np.array([1.0, 0.0]),
-                        weight=np.array([1, weight_fact]),
+                        [1.0, 0.0],
+                        weight=[1, weight_fact],
                         fs=1, type='bandpass', maxiter=50)
 
     # up/down conversion

--- a/tests/python/packages/isce3/signal/fir_filter_func.py
+++ b/tests/python/packages/isce3/signal/fir_filter_func.py
@@ -245,6 +245,9 @@ def test_bandpass(width, fc):
                                       transition_width=tw, force_odd_len=odd)
     # compute frequency response over entire unit circle
     f, H = sig.freqz(h, fs=1.0, whole=True)
+    # Workaround for scipy=1.16.0
+    # (see https://github.com/isce-framework/isce3/pull/49#issuecomment-3025125318).
+    f = f.real
     a, b = transition_region(width, tw)
     # should have unit gain in passband with ripple < -att
     mask_pass = wrapped_distance(f - fc) < a

